File: leetcode/editor/cn/231_2 的幂_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertTrue(self.solution.isPowerOfTwo(1))
        self.assertTrue(self.solution.isPowerOfTwo(16))
        self.assertFalse(self.solution.isPowerOfTwo(15))


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        pivot = 1
        while pivot < n:
            pivot = pivot << 1
        return pivot == n


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/231_2 的幂_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when231_thenSuccess(t *testing.T) {
	assert.Equal(t, true, isPowerOfTwo(1))
	assert.Equal(t, true, isPowerOfTwo(16))
	assert.Equal(t, false, isPowerOfTwo(15))
}

//leetcode submit region begin(Prohibit modification and deletion)
func isPowerOfTwo(n int) bool {
	pivot := 1
	for pivot < n {
		pivot <<= 1
	}
	return pivot == n
}

//leetcode submit region end(Prohibit modification and deletion)
